### PR TITLE
CB-10039 Accept assets URIs for contact photos

### PR DESCRIPTION
**Pull request by vladimir-kotikov**
JIRA issue [CB-10039](https://issues.apache.org/jira/browse/CB-10039)

This is a replacement for #77 that handles URLs and absolute paths more correctly

### CODE REVIEW

File: src/android/ContactAccessorSdk5.java [hunk 1]
```diff
@@ -1631,24 +1631,41 @@ private void insertPhoto(ArrayList<ContentProviderOperation> ops,
...
-       * @return an input stream
+     * Get an input stream based on file path or uri content://, http://, file://
+     *
+     * @param path path to file
+     * @return an input stream
      * @throws IOException
-       */
+     */
     private InputStream getPathFromUri(String path) throws IOException {
+        String ASSET_PREFIX = "file:///android_asset/";
```
Comment (dblotsky) [inline src/android/ContactAccessorSdk5.java]: Please set this as a constant at the top, similarly to `EMAIL_REGEXP ` and `MAX_PHOTO_SIZE `.

**Comment by vladimir-kotikov** [comment]
@dblotsky, updated.

**Comment by dblotsky** [comment]
Thanks! The change looks much cleaner now. How are this PR and #77 related? Will they both be merged in?

**Comment by riknoll** [comment]
We really need to get all this file reading code in one place for Android. The camera plugin does things a bit differently (see [here](https://github.com/apache/cordova-plugin-camera/blob/master/src/android/FileHelper.java)). Is this the correct way to interpret relative paths? This assumes that they should be based on the assets root directory, but should it act like a web page and respect the file structure of the www folder? What do other platforms do? @jasongin thoughts?

**Comment by vladimir-kotikov** [comment]
@dblotsky, this replaces #77

**Comment by vladimir-kotikov** [comment]
@riknoll, other platforms (iOS, Windows) don't try to resolve relative paths. Instead they expect image path/URI to be absolute. Probably we should follow the same rule here. However, we still need to threat `file:///android_asset` URIs differently, so the condition `if (path.startsWith(ASSET_URL_PREFIX)) ... ` is still actual. Thoughts?

**Comment by riknoll** [comment]
@vladimir-kotikov That sounds good to me! I agree we should be handling `file://android_asset` URIs

**Comment by jasongin** [comment]
I agree the file path handling code probably should be centralized in the platform lib for each platform. But it should be done as part of a broader effort to make all plugin APIs more consistent in handling files. (My audit supporting that is still in progress.)

I'm OK with adding this bit of duplicated code for now to fix this bug, knowing it may be refactored later. Also note ideally the contact photo APIs should also support cdvfile:// paths, assuming we don't deprecate cdvfile://. But again I wouldn't block this fix because of that.

**Pull request closed without merging**